### PR TITLE
[tune][dashboard] upgrade tensorboardX to 2.6.0

### LINKED ISSUE #36658: [dashboard] Dashboard Test is failing due to tensorboardX 's protobuf dependency issue.
### What happened + What you expected to happen

https://buildkite.com/ray-project/oss-ci-build-branch/builds/4563#0188de6c-10fd-4fbf-b52f-999140220081

or

https://buildkite.com/ray-project/oss-ci-build-branch/builds/4568#0188df13-0a86-44d2-81a1-86469747982b

Expects test to pass

### Versions / Dependencies

current master

### Reproduction script

open a blank PR that touches dashboard with a noop change, and one should be able to see the test failure.

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by aslonnie**
which unpins protobuf and uses protobuf>=4

fixes https://github.com/ray-project/ray/issues/36658

### CODE REVIEW

**Comment by pcmoritz** [comment]
Thanks for fixing this! Also ouch we are running into dependency hell:
```
The conflict is caused by:
--
  | opentelemetry-exporter-opencensus 0.20b0 depends on protobuf>=3.13.0
  | tensorboardx 2.6.1 depends on protobuf>=4.22.3
  | tensorflow 2.11.0 depends on protobuf<3.20 and >=3.9.2
  | The user requested (constraint) protobuf!=3.19.5,>=3.15.3
```

**Comment by pcmoritz** [comment]
Even Google's own TensorFlow is running into protobuf hell, quite telling:

https://github.com/tensorflow/tensorflow/commit/84f40925e929d05e72ab9234e53c729224e3af38#diff-f526feeafa1000c4773410bdc5417c4022cb2c7b686ae658b629beb541ae9112R100

**Comment by aslonnie** [comment]
test passing: https://buildkite.com/ray-project/oss-ci-build-pr/builds/26248#0188df83-aab2-4e21-b2eb-74ba147c0d47

File: python/requirements.txt [hunk 1]
```diff
@@ -53,7 +53,7 @@ scikit-image
 prometheus_client>=0.7.1
 requests
 pandas
-tensorboardX>=1.9
+tensorboardX<=2.6.0,>=1.9
```
Comment (sihanwang41) [inline python/requirements.txt:56]: can you add a comment why we need to pin? so that we can easily get the context for future upgrading.

Comment (aslonnie) [reply python/requirements.txt]: done.

**Review by can-anyscale — APPROVED** [review]

**Comment by aslonnie** [comment]
there are many other tests on master that are breaking now, so I honestly do not really know if this pr's change is breaking the tests even more..

@scv119 , how is your progress on fixing master?

**Pull request merged**